### ISSUE TITLE
Format async block and async fn

### ISSUE THREAD

**Issue by topecongiro** [rust-lang-nursery/rustfmt#2812]
E.g. 
```rust
async unsafe fn foo() {
    async move {}
}
async fn bar() {}
```

**Comment by rfernbach** [comment]
I would be interested in tackling this.

**Comment by nrc** [comment]
Fixed by #2874

**Comment by reisub0** [comment]
I'm unsure of whether to open a new issue or not. The latest stable version of rustfmt
```
rustfmt 1.4.8-stable (afb1ee1 2019-09-08)
```

fails to format this program
```rust
use async_std::task;

fn main() {
    task::block_on(async {
        println!("Hello, world!");
    })
}
```
With this error:
```rust
error: expected one of `,` or `}`, found `!`
 --> <stdin>:5:16
  |
4 |     task::block_on(async {
  |                    ----- while parsing this struct
5 |         println!("Hello, world!");
  |                ^ expected one of `,` or `}` here

```
The same is true for the latest nightly
```
rustfmt 1.4.9-nightly (33e3667 2019-10-07)
```

Am I doing something wrong?

**Comment by calebcartwright** [comment]
@reisub0 - are you using `cargo fmt` or running `rustfmt` directly? 

If the latter, try including the edition flag `--edition 2018`

**Comment by reisub0** [comment]
@calebcartwright - Thank you! Using the `--edition` flag worked. I was using `rustfmt` on <stdin> mode as part of a (Neo)Vim plugin and not `cargo fmt`. I expected the default edition to be the latest(2018). And I didn't find any other documentation on the README. But I guess it hasn't been merged yet as that was only a recent change (#3942).

Thanks for your help and sorry for the bother.

**Comment by calebcartwright** [comment]
No worries @reisub0, happy to help! Also, if you ever have any future questions, another available option is the Discord channel for rustfmt (this can be especially useful if you're unsure whether to create a new issue)
